proposals/lip-0055.md

```diff
@@ -599,16 +609,22 @@ def verifyEventRoot():
 
 Here, the function `eventRoot` calculates the events root of the block with the input height as specified in [LIP 0065][lip-0065].
 
-##### Max Height Prevoted and Max Height Generated
+##### Max Height Prevoted, Max Height Generated and Implies Max Prevotes
 
-The properties `maxHeightPrevoted` and `maxHeightGenerated` are related to the [Lisk-BFT protocol][lip-0014#block-header-properties]. They are verified by calling the `verifyBFTProperties` function. This function returns a boolean, indicating the success of the check.
+The properties `maxHeightPrevoted`, `maxHeightGenerated` and `impliesMaxPrevotes` are related to the [Lisk-BFT protocol][lip-0014#block-header-properties]. They are verified by calling the `verifyBFTProperties` function. This function returns a boolean, indicating the success of the check.
 
 ```python
 def verifyBFTProperties():
   if block.header.maxHeightPrevoted != bft.getBFTHeights().maxHeightPrevoted:
     return False
 
-  return not bft.isHeaderContradictingChain(block.header)
+  if bft.isHeaderContradictingChain(block.header):
+    return False
+
+  if block.header.impliesMaxPrevotes != bft.impliesMaximalPrevotes(block.header)
```

Comment:
Isn't the whole function `verifyBFTProperties()` part of the engine now so all information would be available?

Reply:
yes, all the information is available. I guess I just need to slightly modify the `impliesMaximalPrevotes` function to handle genesis block case, but it can be handled.